ocaml/libs/xapi-stdext/lib/xapi-stdext-threads/ipq.ml

```diff
@@ -64,10 +60,20 @@ let maximum h =
 
 let remove h s =
   if h.size <= 0 then raise EmptyHeap ;
+  if s < 0 || s >= h.size then invalid_arg "index out of bounds" ;
```

Comment:
Might want to use `__FUNCTION__` or `__LOC__` in this message. I generally like to define a function to raise an exception that takes printf-style arguments for convenience (using `Printf.ksprintf`).

Reply:
I copied the standard message causing out of index using a standard `Array` which was consistent with the error if you passed an index out of the data array. Can you suggest a message and how to change the test accordingly?

Reply:
I would at least include __FUNCTION__ in the message to avoid guessing where it is emerging from. 
```
invalid_arg (Printf.sprintf "%s: index %d out of bounds" __FUNCTION__ s)
```